test/helpers/svr_in_proc.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require_relative 'tmp_path'
+require_relative 'svr_base'
+require_relative 'sockets'
+
+require "puma/configuration"
+require "puma/events"
+# avoids circular require when running single tests
+require 'nio'
+
+module TestPuma
+  class SvrInProc < SvrBase
+    include TmpPath
+    include TestPuma::Sockets
+
+    def teardown
+      return if skipped?
+      if @server
+        @server.stop true
+        @server.binder.close_listeners
+      end
+
+      if @ios_to_close
+        @ios_to_close.each do |io|
+          io.close if io.is_a?(IO) && !io.closed?
+          io = nil
+        end
+      end
+
+      if defined?(@bind_port) && @bind_port
+        # ssl sockets need extra time to close?
```

Comment:
Oof, that's a long sleep...

Reply:
Let me check that....

Reply:
I checked this, and can't find a reason for it.  But, there aren't any tests using `svr_in_proc.rb` with ssl connections.